pint_server/app.py

```diff
@@ -425,11 +439,35 @@ def get_provider_servers(provider):
             for server in servers]
 
 
+def trim_images_payload(images):
+    payload_size = jsonify(images=images).content_length
+
+    if payload_size >  MAX_PAYLOAD_SIZE:
+        # NOTE: assuming the size of the entries are evenly distributed, we
+        # determine the percentage of entries to trim from the end of the list,
+        # as the list is sorted in decending order by publishedon date, by
+        # calculating the percentage over the maximum payload size. Then
+        # trim the same percentage off the list, rounding up to be safe.
+        trim_size  = math.ceil(
+            ((payload_size - MAX_PAYLOAD_SIZE) / payload_size) * len(images))
+        images = images[:-trim_size]
+
+        # Now make sure we don't have partial data by finished triming all the
+        # images from all regions that have the same publishedon date that of
+        # the last image that got trimmed.
+        publishedon = images[-1]['publishedon']
+        while images[-1]['publishedon'] == publishedon:
+            images.pop()
+    return images
+
+
 def get_provider_images(provider):
-    images = PROVIDER_IMAGES_MODEL_MAP[provider].query.all()
+    images = PROVIDER_IMAGES_MODEL_MAP[provider].query.order_by(
+        desc(PROVIDER_IMAGES_MODEL_MAP[provider].publishedon)).all()
     exclude_attrs = PROVIDER_IMAGES_EXCLUDE_ATTRS.get(provider)
-    return [get_formatted_dict(image, exclude_attrs=exclude_attrs)
-            for image in images]
+    return trim_images_payload(
```

Comment:
So with this change all `/v1/<provider>/images` queries will be trimmed, no matter whether XML or JSON was requested... Is that OK?

Reply:
Per discussion in chat, this is intentional/desirable as we want the XML and JSON result to be equivalent.